eth/client.go

```diff
@@ -418,7 +419,184 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	// Get total bonded
+	totalBonded, err := c.TranscoderTotalStake(delegator.DelegateAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	newStake := totalBonded.Sub(totalBonded, amount)
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	hints := simulateTranscoderPoolUpdate(delegator.DelegateAddress, newStake, transcoders, isFull)
+
+	return c.UnbondWithHint(amount, hints.PosPrev, hints.PosNext)
+}
+
+func (c *client) RebondFromUnbonded(to ethcommon.Address, unbondingLockID *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	totalBonded, err := c.TranscoderTotalStake(to)
+	if err != nil {
+		return nil, err
+	}
+
+	lock, err := c.GetDelegatorUnbondingLock(sender, unbondingLockID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get unbonding lock")
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	newStake := totalBonded.Add(totalBonded, lock.Amount)
+
+	hints := simulateTranscoderPoolUpdate(to, newStake, transcoders, isFull)
+
+	return c.RebondFromUnbondedWithHint(to, unbondingLockID, hints.PosPrev, hints.PosNext)
+}
+
+func (c *client) Rebond(unbondingLockID *big.Int) (*types.Transaction, error) {
+	sender := c.Account().Address
+
+	// Get delegator
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	currentRound, err := c.CurrentRound()
+	if err != nil {
+		return nil, err
+	}
+
+	totalBonded, err := c.PendingStake(sender, currentRound)
+	if err != nil {
+		return nil, err
+	}
+
+	lock, err := c.GetDelegatorUnbondingLock(sender, unbondingLockID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get unbonding lock")
+	}
+
+	transcoderStake, err := c.TranscoderTotalStake(delegator.DelegateAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	newStake := transcoderStake.Add(totalBonded, lock.Amount)
```

Comment:
We don't need to include the delegator's pending stake here since that is already accounted for in the transcoder's total stake.